### PR TITLE
wget 1.19.1

### PR DESCRIPTION
**Pull request by zmwangx**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

- Make `openssl` mandatory and `libidn` recommended per discussions in #9613;
- Test: write retrieved page to `/dev/null` (the hundreds of kilobytes of output on stdout is useless).

### CODE REVIEW

File: Formula/wget.rb [hunk 1]
```diff
@@ -23,16 +23,15 @@ class Wget < Formula
...
-  option "with-iri", "Enable iri support"
   option "with-debug", "Build with debug support"
+  option "without-libidn", "Build without IRI support"
 
   depends_on "pkg-config" => :build
   depends_on "pod2man" => :build if MacOS.version <= :snow_leopard
-  depends_on "openssl" => :recommended
-  depends_on "libidn" if build.with? "iri"
+  depends_on "openssl"
+  depends_on "libidn" => :recommended
```
Comment (MikeMcQuaid) [inline Formula/wget.rb]: What does `libidn` do? Not sure what IRI support is but I'd personally rather avoid the extra dependency given how infrequently this option is specified: 

screen shot 2017-02-11 at 21 32 01

**Comment by zmwangx** [comment]
@MikeMcQuaid [Internationalized domain name](https://en.wikipedia.org/wiki/Internationalized_domain_name).

"IRI support" is somewhat confusing because Wget can handle Unicode in path, query and fragment (which is simply a matter of percent-encoding) just fine without libidn; however, it requires libidn to support Unicode characters in the domain name.

As I said in https://github.com/Homebrew/homebrew-core/pull/9613#issuecomment-277555936, I'm recommending libidn on principle: 1. Wget defaults to supporting IDNs (when libidn is available, that is); 2. The web is increasingly internationalized; I've certainly encountered IDNs in the wild, and I expect to see more. (I do realize you can translate IDNs elsewhere and feed the ASCII version to a Wget without IDN support, e.g., copying from browser's address bar.)

However, I just realized that Wget 1.19 dropped support for libidn in favor of libidn2 (commit https://github.com/bettermirror/wget/commit/00ae9b4ee2caa6640776e9777e7ef5a374d082c0). Therefore, I removed libidn-related code from the formula.

Someone could contribute a libidn2 formula to bring the option back.

**Comment by ilovezfs** [comment]
Also probably important to note the security concerns with idn: https://en.wikipedia.org/wiki/Internationalized_domain_name#ASCII_spoofing_concerns

>The use of Unicode in domain names makes it potentially easier to spoof web sites as the visual representation of an IDN string in a web browser may make a spoof site appear indistinguishable to the legitimate site being spoofed, depending on the font used. For example, Unicode character U+0430, Cyrillic small letter a, can look identical to Unicode character U+0061, Latin small letter a, used in English. As a concrete example, using Cyrillic letters а, е (“Ie”/“Ye”, U+0435, looking essentially identical to Latin letter e), Belarusian-Ukrainian і (U+0456, essentially identical to Latin letter i), р (“Er”, U+0440, essentially identical to Latin letter p), we form the URL wіkіреdіа.org (xn--wkd-8cdx9d7hbd.org in encoded form), which is virtually indistinguishable from the visual representation of the legitimate wikipedia.org (possibly depending on fonts).

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**